packages/ember-data/lib/system/relationship-meta.js

```diff
@@ -6,7 +6,7 @@ export function typeForRelationshipMeta(store, meta) {
   typeKey = meta.type || meta.key;
```

Comment:
I might say `typeKey` here is inappropriate and misleading. This is `userEnteredSomeKey`